### PR TITLE
Implement start drag for WPF browser

### PR DESCRIPTION
**Pull request by mmamone**
Resolves https://github.com/cefsharp/CefSharp/issues/1154
Exposed the necessary functions for start drag to work:
* StartDragging (RenderClientAdapter)
* OnDragSourceEndedAt (ManagedCefBrowserAdapter)
* OnDragSourceSystemDragEnded (ManagedCefBrowserAdapter)

Implemented IRenderWebBrowser.StartDragging for the WPF
ChromiumWebBrowser.

TODO: Would like to add an adorner layer for the dragged text (using the
supplied fragment html), but feels like a project for later.

### CODE REVIEW

**Comment by mmamone** [comment]
Feel free to rip this PR to shreds. This is my first contribution to an open source project ( :dancer: ).

**Comment by mmamone** [comment]
Hmmm... Hold off on this one. I'll update it to use a DataObject to contain the copy data.

**Comment by amaitland** [comment]
@mmamone Great first contribution thanks :+1:

File: CefSharp.Core/Internals/RenderClientAdapter.h [hunk 1]
```diff
@@ -177,6 +177,13 @@ namespace CefSharp
                 _renderWebBrowser->SetCursor((IntPtr)cursor, (CefSharp::CefCursorType)type);
             };
 
+            virtual DECL bool StartDragging(CefRefPtr<CefBrowser> browser, CefRefPtr<CefDragData> drag_data,
```
Comment (amaitland) [inline CefSharp.Core/Internals/RenderClientAdapter.h]: Trying to slowly bring get consistency with naming. e.g. `drag_data` becomes `dragData`

**Comment by amaitland** [comment]
@mmamone Just a heads up that `GitHub` doesn't send out any notification when you push more commits, so generally advisable to post a comment when ready.

Can you also provide a quick summary of the testing you've done?

**Comment by mmamone** [comment]
I did the following for my basic testing :
* Drag drop within the application (using the test application)
* Drag drop between applications (parity with the CEF test application)
 * Can drop text to basic text editors
 * Dragging links seems to work
 * Dragging images doesn't provide a link
 * Drop into word doesn't seem to work (though visual studio worked)

Let me know if there are any other scenarios I should try out. I'm guessing things like use cases don't exist ?

**Comment by amaitland** [comment]
> I'm guessing things like use cases don't exist ?

Have enough trouble getting people to contribute as it is, let alone add use cases!

Will merge this `ASAP`. Thanks :+1:

**Comment by mmamone** [comment]
Awesome.
When I get more time, may have a go at adding an adorner layer. I was able to get a drag visual going, but displaying the selected text in the visual was proving nightmarish.

**Comment by tomby** [comment]
it's meaningful
l to me

**Pull request merged**